.github/workflows/node.js.yaml

```diff
@@ -1,57 +1,85 @@
 name: NodeJS CI
 
 on:
-    push:
-        branches: ["master", "develop"]
-    pull_request:
-        branches: ["master", "develop"]
+  push:
+    branches: [ "master", "develop" ]
+  pull_request:
+    branches: [ "master", "develop" ]
 
 jobs:
-    build:
-        runs-on: ubuntu-latest
+  build:
+    runs-on: ubuntu-latest
 
-        strategy:
-            matrix:
-                # See https://nodejs.org/en/about/releases/
-                node-version: [16.x]
+    strategy:
+      matrix:
+        # See https://nodejs.org/en/about/releases/
+        node-version: [ 16.x ]
 
+    env:
+      SOL_AST_COMPILER_CACHE: ${{ github.workspace }}/.compiler_cache
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Use NodeJS ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: "npm"
+
+      - name: Install and lint
+        run: |
+          npm install
+          npm link
+          npm run lint
+
+      # See https://github.com/ethereum/solc-bin
+      # See https://binaries.soliditylang.org/
+      # Also, remove list files and one rarely used compiler to still test downloading on-demand.
+      - name: Pre-download compilers from historical builds archive
+        run: |
+          scribble --download-compilers native wasm
+          find $SOL_AST_COMPILER_CACHE -name 'list.json' -delete
+          find $SOL_AST_COMPILER_CACHE -name '*v0.5.17*' -delete
+
+      - name: Test and generate coverage report
+        run: |
+          npm run test:ci
+          npm run coverage
+
+      - name: Upload coverage to CodeCov
+        uses: codecov/codecov-action@v3
+        with:
+          name: codecov-umbrella
+          directory: ./coverage/
+          fail_ci_if_error: true
+          verbose: true
+
+  notify-slack:
+    needs: build
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/master'
```

Comment:
Can we also extend it to inform about merges to `develop`? We use two branches in this repository.

Reply:
removed the if statement and modified the env to pick the branch name